src/Admin/Pap/CampaignHistoryAdmin.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace App\Admin\Pap;
+
+use App\Pap\CampaignHistoryStatusEnum;
+use App\Pap\CampaignHistoryVoterStatusEnum;
+use Sonata\AdminBundle\Admin\AbstractAdmin;
+use Sonata\AdminBundle\Datagrid\DatagridMapper;
+use Sonata\AdminBundle\Datagrid\ListMapper;
+use Sonata\AdminBundle\Route\RouteCollection;
+use Sonata\DoctrineORMAdminBundle\Filter\ChoiceFilter;
+use Sonata\DoctrineORMAdminBundle\Filter\DateRangeFilter;
+use Sonata\DoctrineORMAdminBundle\Filter\ModelAutocompleteFilter;
+use Sonata\Form\Type\DateRangePickerType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+
+class CampaignHistoryAdmin extends AbstractAdmin
+{
+    protected function configureRoutes(RouteCollection $collection)
+    {
+        $collection->clearExcept('list');
+    }
+
+    protected function configureDatagridFilters(DatagridMapper $filter)
+    {
+        $filter
+            ->add('campaign', null, [
+                'label' => 'Campagne',
+                'show_filter' => true,
+            ])
+            ->add('dataSurvey.survey', null, [
+                'label' => 'Questionnaire',
+                'show_filter' => true,
+            ])
+            ->add('questioner', ModelAutocompleteFilter::class, [
+                'label' => 'Militant',
+                'show_filter' => true,
+                'field_options' => [
+                    'property' => [
+                        'firstName',
+                        'lastName',
+                    ],
+                ],
+            ])
+            ->add('adherent', ModelAutocompleteFilter::class, [
+                'label' => 'Contacté',
+                'field_options' => [
+                    'property' => [
+                        'firstName',
+                        'lastName',
+                    ],
+                ],
+            ])
+            ->add('beginAt', DateRangeFilter::class, [
+                'label' => 'Date de début',
+                'field_type' => DateRangePickerType::class,
+            ])
+            ->add('status', null, [
+                'label' => 'Statut',
+                'field_type' => ChoiceType::class,
+                'field_options' => [
+                    'choices' => array_flip(CampaignHistoryStatusEnum::LABELS),
+                ],
+            ])
+            ->add('type', ChoiceFilter::class, [
+                'label' => 'Inscrit sur les listes',
+                'field_type' => ChoiceType::class,
+                'field_options' => [
+                    'choices' => array_flip(CampaignHistoryVoterStatusEnum::LABELS),
+                ],
+            ])
+        ;
+    }
+
+    protected function configureListFields(ListMapper $listMapper)
+    {
+        $listMapper
+            ->add('campaign', null, [
+                'label' => 'Campagne',
+            ])
+            ->add('building.address', null, [
+                'label' => 'Adresse',
+                'template' => 'admin/pap/campaign_history/list_address.html.twig',
```

Comment:
what about using the existing template with a little helper:
```php
CampaignHistory::getAddress() {
  return $this->building->getAddress();
}
``` 
```suggestion
                'template' => 'admin/list_address.html.twig',
```

Reply:
This template is for another entity `Address`, it does not match the fields of `PapAddress`.